recipes/boost/all/test_package/conanfile.py

```diff
@@ -1,94 +1,101 @@
+from conan import ConanFile
+from conan.errors import ConanException
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.scm import Version
+from conans import tools as tools_legacy
 import os
 
-from conan.tools.build import cross_building
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanException
-
 
 class TestPackageConan(ConanFile):
-    settings = "os", "compiler", "arch", "build_type"
-    generators = "cmake", "cmake_find_package"
-
-    def build_requirements(self):
-        if self.settings.os == "Macos" and self.settings.arch == "armv8":
-            # Attempting to use @rpath without CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG being
-            # set. This could be because you are using a Mac OS X version less than 10.5
-            # or because CMake's platform configuration is corrupt.
-            self.build_requires("cmake/3.20.1")
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps", "VirtualRunEnv"
+    test_type = "explicit"
 
     def _boost_option(self, name, default):
         try:
             return getattr(self.options["boost"], name, default)
         except (AttributeError, ConanException):
             return default
 
+    def layout(self):
+        cmake_layout(self)
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["HEADER_ONLY"] = self.dependencies["boost"].options.header_only
+        if not self.dependencies["boost"].options.header_only:
```

Comment:
This option does not like me either

```log
boost/1.80.0 (test package): Calling build()
boost/1.80.0 (test package): CMake command: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/Users/christopherm/conan-center-index/recipes/boost/all/test_package/build/generators/conan_toolchain.cmake" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/christopherm/conan-center-index/recipes/boost/all/test_package/."
-- Using Conan toolchain: /Users/christopherm/conan-center-index/recipes/boost/all/test_package/build/generators/conan_toolchain.cmake
-- Conan: Component target declared 'Boost::headers'
-- Conan: Component target declared 'Boost::boost'
-- Conan: Target declared 'boost::boost'
-- Configuring done
CMake Error at CMakeLists.txt:12 (target_link_libraries):
  Target "random_exe" links to:

    Boost::random

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at CMakeLists.txt:18 (target_link_libraries):
  Target "regex_exe" links to:

    Boost::regex

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at CMakeLists.txt:24 (target_link_libraries):
  Target "test_exe" links to:

    Boost::unit_test_framework

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at CMakeLists.txt:30 (target_link_libraries):
  Target "coroutine_exe" links to:

    Boost::coroutine

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at CMakeLists.txt:36 (target_link_libraries):
  Target "chrono_exe" links to:

    Boost::chrono

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at CMakeLists.txt:63 (target_link_libraries):
  Target "locale_exe" links to:

    Boost::locale

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at CMakeLists.txt:70 (target_link_libraries):
  Target "stacktrace_addr2line_exe" links to:

    Boost::stacktrace_addr2line

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at CMakeLists.txt:76 (target_link_libraries):
  Target "stacktrace_backtrace_exe" links to:

    Boost::stacktrace_backtrace

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at CMakeLists.txt:84 (target_link_libraries):
  Target "stacktrace_noop_exe" links to:

    Boost::stacktrace_noop

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at CMakeLists.txt:97 (target_link_libraries):
  Target "stacktrace_basic_exe" links to:

    Boost::stacktrace_basic

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
ERROR: boost/1.80.0 (test package): Error in build() method, line 55
	cmake.configure()
	ConanException: Error 1 while executing cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/Users/christopherm/conan-center-index/recipes/boost/all/test_package/build/generators/conan_toolchain.cmake" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/christopherm/conan-center-index/recipes/boost/all/test_package/."
 ✘ christopherm@christopherm-mac  ~/conan-center-index/recipes/boost   boost-conan-v2  conan create all boost/1.80.0@ -o boost:header_only=True
```

Reply:
`boost:header_only` option is supposed to be tested in c3i, so I don't understand how it can fail on your machine.

Reply:
Maybe you have not cleared build folder and CMakeUserPresets of test package after a conan create with another set of options, it's a common mistake when you test conan v2 generators in test package.